src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/ECDsa/ECDsaImportExport.cs

```diff
@@ -0,0 +1,297 @@
+                Assert.False(verified);
+            }
+        }
+
+        [Theory, MemberData(nameof(TestInvalidCurves))]
+        public static void TestExplicitCurvesNegative(CurveDef curveDef)
+        {
+            // An exception may be thrown during Create() if the Oid is bad, or later during native calls
+            Assert.Throws<PlatformNotSupportedException>(() => ECDsaFactory.Create(curveDef.Curve).ExportExplicitParameters(false));
+        }
+
+        [Fact]
+        public static void TestNamedCurveNegative()
+        {
+            Assert.Throws<ArgumentNullException>(() => ECCurve.CreateFromFriendlyName(null));
+            Assert.Throws<ArgumentNullException>(() => ECCurve.CreateFromValue(null));
+            Assert.Throws<ArgumentException>(() => ECCurve.CreateFromFriendlyName(""));
+            Assert.Throws<PlatformNotSupportedException>(() => ECDsaFactory.Create(ECCurve.CreateFromFriendlyName("Invalid")).ExportExplicitParameters(false));
+            Assert.Throws<ArgumentException>(() => ECCurve.CreateFromValue(""));
+            Assert.Throws<PlatformNotSupportedException>(() => ECDsaFactory.Create(ECCurve.CreateFromValue("Invalid")).ExportExplicitParameters(false));
+            Assert.Throws<ArgumentException>(() => ECCurve.CreateFromOid(new Oid(null, null)));
+            Assert.Throws<ArgumentException>(() => ECCurve.CreateFromOid(new Oid("", "")));
+        }
+
+        [Fact]
+        public static void TestKeySizeCreateKey()
+        {
+            using (ECDsa ec = ECDsaFactory.Create(ECCurve.NamedCurves.nistP256))
+            {
+                // Ensure the handle is created
+                Assert.Equal(256, ec.KeySize);
+                ec.Exercise();
+
+                ec.KeySize = 521; //nistP521
+                Assert.Equal(521, ec.KeySize);
+                ec.Exercise();
+
+                Assert.ThrowsAny<CryptographicException>(() => ec.KeySize = 9999);
+            }
+        }
+
+        [ConditionalFact(nameof(ECExplicitCurvesSupported))]
+        public static void TestExplicitImportValidationNegative()
+        {
+            unchecked
+            {
+                using (ECDsa ec = ECDsaFactory.Create())
+                {
+                    ECParameters p = ECDsaTestData.GetNistP256ExplicitTestData();
+                    Assert.True(p.Curve.IsPrime);
+                    ec.ImportParameters(p);
+
+                    ECParameters temp = p;
+                    temp.Q.X = null; Assert.ThrowsAny<CryptographicException>(() => ec.ImportParameters(temp));
+                    temp.Q.X = new byte[] { }; Assert.ThrowsAny<CryptographicException>(() => ec.ImportParameters(temp));
+                    temp.Q.X = new byte[1] { 0x10 }; Assert.ThrowsAny<CryptographicException>(() => ec.ImportParameters(temp));
+                    temp.Q.X = (byte[])p.Q.X.Clone(); --temp.Q.X[0]; Assert.ThrowsAny<PlatformNotSupportedException>(() => ec.ImportParameters(temp));
```

Comment:
On the cng side, it is somewhat ambiguous when a not_supported error comes back as to whether the curve is not supported, or one of the parameters. I've tried to throw PNSE when the curve is invalid, but that check is not 100% unless we specifically ask first by obtaining the list of supported curves which we don't do. Not sure how the openssl and cng sides started differing here w.r.t. exceptions and the tests were consistent last week (the last time I ran them locally). I will investigate here